sos/plugins/maas.py

```diff
@@ -53,18 +53,48 @@ def _remote_api_login(self):
         return ret['status'] == 0
 
     def setup(self):
-        self.add_copy_spec([
-            "/etc/squid-deb-proxy",
-            "/etc/maas",
-            "/var/lib/maas/dhcp*",
-            "/var/log/apache2*",
-            "/var/log/maas*",
-            "/var/log/upstart/maas-*",
-        ])
-        self.add_cmd_output([
-            "apt-cache policy maas-*",
-            "apt-cache policy python-django-*",
-        ])
+        snap_list_output = self.exec_cmd('snap list maas')
+        if snap_list_output["status"] == 0:
+            # We're in the snap, so file paths are different, and all
+            # the related software (i.e. bind, dhcpd) are configured
+            # and maintained by the snap
+            self.add_cmd_output([
+                'snap info maas',
+                'maas status'])
+            # Don't send secrets
+            self.add_forbidden_path([
+                "/var/snap/maas/current/bind/session.key"
+            ])
+            self.add_copy_spec([
+                "/var/snap/maas/common/log",
+                "/var/snap/maas/common/snap_mode",
+                "/var/snap/maas/current/*.conf",
+                "/var/snap/maas/current/bind",
+                "/var/snap/maas/current/http",
+                "/var/snap/maas/current/supervisord",
+                "/var/snap/maas/current/preseeds",
+                "/var/snap/maas/current/proxy",
+                "/var/snap/maas/current/rsyslog",
+            ])
+            # As of 2.7.0 the only systemd service that is in the snap
+            # is supervisor, the processes that it maintains are shown
+            # in the output of `maas status`.
+            self.services = (
+                'snap.maas.supervisor',
+            )
```

Comment:
That was my first thought as well that plugin enablement makes no sense here :) Imho the purpose is collecting the services status and journal logs few lines below : https://github.com/sosreport/sos/blob/master/sos/plugins/maas.py#L69-L72 . But is is quite confusing..

Reply:
Sure but the `services` member has its own meaning - since this reference is only used in the scope of `setup()` it's confusing to also set the class member at this point.

Reply:
Right, I didn't want the plugin to attempt to get status of services that don't exist in the snap world.

Of course this adds more to the argument that it should be a separate plugin